src/main/java/fi/aalto/cs/apluscourses/utils/FileDateFormatter.java

```diff
@@ -0,0 +1,86 @@
+package fi.aalto.cs.apluscourses.utils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileTime;
+import java.time.Clock;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+public class FileDateFormatter {
```

Comment:
I think there are two things going on here: formatting the date and getting the last modification time of a file. I could easily see the date formatting being useful in general also for other dates that don't come from files.

I would prefer it if we had a generic `DateFormatter` that does the formatting (regardless of whether the date comes from a file or something else). The `getFileModificationTime` would then either be a convenience method of `DateFormatter`, or `Files.getLastModifiedTime(...)` could be moved to `SubmissionViewModel`, or something else could work as well.

Reply:
Yeah, I didn't see this comment but addressed the same thing in [my comment](https://github.com/Aalto-LeTech/intellij-plugin/pull/472#pullrequestreview-568367600).